test/acceptance/tests/ingress-gateway/ingress_gateway_test.go

```diff
@@ -0,0 +1,192 @@
+package connect
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/k8s"
+	"github.com/hashicorp/consul-helm/test/acceptance/framework"
+	"github.com/hashicorp/consul-helm/test/acceptance/helpers"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// Test that ingress gateways work in a default installation.
+func TestIngressGatewayDefault(t *testing.T) {
+	env := suite.Environment()
+
+	// Initially we start without ingress gateways enabled because we need
+	// to create their config entry first.
+	helmValues := map[string]string{
+		"connectInject.enabled": "true",
+	}
+
+	releaseName := helpers.RandomName()
+	consulCluster := framework.NewHelmCluster(t, helmValues, env.DefaultContext(t), suite.Config(), releaseName)
+
+	consulCluster.Create(t)
+
+	t.Log("creating server")
+	createServer(t, suite.Config(), env.DefaultContext(t).KubectlOptions())
+
+	// We use a "bounce" pod so that we can make calls to the ingress gateway
+	// via kubectl exec without needing a route into the cluster from the test machine.
+	t.Log("creating bounce pod")
+	createBouncePod(t, suite.Config(), env.DefaultContext(t).KubectlOptions())
+
+	// With the cluster up, we can create our ingress-gateway config entry.
+	t.Log("creating config entry")
+	consulClient := consulCluster.SetupConsulClient(t, false)
+
+	// Create config entry
+	created, _, err := consulClient.ConfigEntries().Set(&api.IngressGatewayConfigEntry{
+		Kind: api.IngressGateway,
+		Name: "ingress-gateway",
+		Listeners: []api.IngressListener{
+			{
+				Port:     8080,
+				Protocol: "tcp",
+				Services: []api.IngressService{
+					{
+						Name: "static-server",
+					},
+				},
+			},
+		},
+	}, nil)
+	require.NoError(t, err)
+	require.Equal(t, true, created, "config entry failed")
+
+	// Now we can upgrade the cluster and enable ingress gateways.
+	t.Log("upgrading helm release with ingress gateways enabled")
+	consulCluster.Upgrade(t, map[string]string{
+		"ingressGateways.enabled":              "1",
+		"ingressGateways.gateways[0].name":     "ingress-gateway",
+		"ingressGateways.gateways[0].replicas": "1",
+	})
+
+	// With the ingress gateway up, we test that we can make a call to it
+	// via the bounce pod. It should route to the static-server pod.
+	t.Log("trying calls to ingress gateway")
+	k8sClient := env.DefaultContext(t).KubernetesClient(t)
+	k8sOptions := env.DefaultContext(t).KubectlOptions()
```

Comment:
These aren't used anywhere else? are you extracting them into variables for brevity/readibility?

Reply:
I think it was copy pasta but now that I've merged the defaults and secure tests it's needed since we call checkConnection twice now.

Reply:
makes sense.